Automattic/sensei
# Extend standard YouTube embed block

### Changes proposed in this Pull Request

* Initialize the YouTube player in JavaScript and handle events to implement Video-course progression settings.

### Testing instructions

* Enable the feature flag: `add_filter( 'sensei_feature_flag_video_based_course_progression', '__return_true' );`
* Run `npm start`
* Enable all video settings for a course.
* Add a YouTube video to the course lesson.
* Enroll in the course.
* Open the lesson with a YouTube video.
* Watch the video completely and wait for the auto-submission of the form.
* Disable autocomplete setting and open the lesson again.
* Try to complete the lesson before you watched it completely. (It is not visible yet that the button is disabled.) Shouldn't submit the form.
* Watch the video completely and wait for the auto-submission of the form. Should work.
* Test of auto-pause:
  * Start playing video
  * Try to change tab/window
  * Video should be paused
